python/mxnet/base.py

```diff
@@ -273,18 +273,31 @@ class MXCallbackList(ctypes.Structure):
         ]
 
 
+# pylint: disable=line-too-long
 def _load_lib():
     """Load library by searching possible path."""
     lib_path = libinfo.find_lib_path()
-    if sys.version_info >= (3, 8) and os.name == "nt":
-        # use LOAD_WITH_ALTERED_SEARCH_PATH, For simplicity, let's just fill the numbers.
-        # pylint: disable=E1123
-        lib = ctypes.CDLL(lib_path[0], winmode=0x00000008)
+    try:
+        if sys.version_info >= (3, 8) and os.name == "nt":
+            # use LOAD_WITH_ALTERED_SEARCH_PATH, For simplicity, let's just fill the numbers.
+            # pylint: disable=E1123
+            lib = ctypes.CDLL(lib_path[0], winmode=0x00000008)
+        else:
+            lib = ctypes.CDLL(lib_path[0], ctypes.RTLD_LOCAL)
+        # DMatrix functions
+        lib.MXGetLastError.restype = ctypes.c_char_p
+    except OSError as e:
+        if "libcudnn" in e.args[0]:
```

Comment:
Another possible missing library here is libquadmath. gfortran depends on quadmath, but quadmath is GPL and we can't redistribute it. Thus, let's include a message here that explains to the user that they should manually install libquadmath from their distribution (ie use apt install or yum install, etc.)